### ISSUE TITLE
Bug: Reverse proxy certificates must be copied over to rport server in order to have noVNC working correctly

### ISSUE THREAD

**Issue by korshakov** [cloudradar-monitoring/rport#523]
When rport server in installed behind reverse proxy noVNC web based remote connection will not work because certificates are mismatching with reverse proxy certs. You have to copy over certs from reverse proxy to the cert folder of rport and it works just fine, but the problem is that certs are usually auto-renewed every 3 months meaning i have to do this every 3 months.

Also to have this working properly behind reverse proxy you must have--fqdn value same as the one you are creating in reverse proxy on custom api and server port.

**Comment by thorstenkramm** [comment]
We are working on re-using the main HTTPS port for all web-based tunnel connections. That means, NoVNC and Web-based RDP will not open a random port. They will use the already open API https port. Then SSL offloading happens only on the reverse proxy. 

As an immediate solution, look at the [renewal-hooks](https://eff-certbot.readthedocs.io/en/stable/using.html#id27) of `certbot`. You can put some rsync command in `/etc/letsencrypt/renewal-hooks/post` of your reverse proxy, so certs are automatically copied to the rport server.

**Comment by DwijadasDey** [comment]
Is it just copying the certificates from reverse proxy folder to rport ssl folder OR overwriting rport ssl files ?

Following are the files from Rport SSL folder:

```
rport.mydomain.cloud-ca.key
rport.mydomain.cloud_certificate.pem
rport.mydomain.cloud.csr
rport.mydomain.cloud.ext
rport.mydomain.cloud_privkey.pem
```

If i need to overwrite then i guess following are the mappings for files from lets encrypt  -> Rport SSL

```
 cert.pem -> rport.mydomain.cloud_certificate.pem
 chain.pem -> rport.mydomain.cloud-ca.key 
 fullchain.pem -> rport.mydomain.cloud.csr
 privkey.pem -> port.mydomain.cloud_privkey.pem
```

Please correct me if i am wrong.

Thanks

**Comment by thorstenkramm** [comment]
@DwijadasDey To keep it simple, download the certificates to a new folder. Don't mess with the default folders. 
Open the `rportd.conf` file and change the path of the reverse proxy certificates there.
This allows you to set up either a rsync cron that keeps the certificates updated or, better, create a Let's encrypt hook that copies the certs only when renewed. 

This issue will be closed because it's not a bug. It might appear cumbersome to copy the certificates, but RPort works as expected and flawless.

**Comment by vveerar1** [comment]
Hi, I followed Awesome Open Source's [video](https://www.youtube.com/watch?v=Xc_5qE7ZyYA) to set up an LXC container for RPort. I am trying to place RPort behind reverse proxy. I copied the certificates from my reverse proxy and changed the path in `rportd.conf` file.  The application works with the new certificate locally on `https://localhost:5000/` but not accessible outside on `https://rport.subdomain.com`. Am I missing something? 

PS: I am using reverse proxy using SWAG docker container with a Cloudflare tunnel.
